main.tf

```diff
@@ -103,102 +105,71 @@ resource "aws_cloudfront_distribution" "cloudfront_distribution" {
   }
 
   dynamic "default_cache_behavior" {
-    for_each = [for i in var.dynamic_default_cache_behavior : {
-      allowed_methods             = i.allowed_methods
-      cached_methods              = i.cached_methods
-      target_origin_id            = i.target_origin_id
-      compress                    = i.compress
-      query_string                = i.query_string
-      cookies_forward             = i.cookies_forward
-      headers                     = i.headers
-      viewer_protocol_policy      = i.viewer_protocol_policy
-      min_ttl                     = i.min_ttl
-      default_ttl                 = i.default_ttl
-      max_ttl                     = i.max_ttl
-      lambda_function_association = lookup(i, "lambda_function_association", null)
-    }]
```

Comment:
I noticed you removed all of this in many sections in favor of the syntax thats on line 108.  I am interested to see if this works, and with multiple resources as this would clean up the file a bit

Reply:
it should work, tested all my changes locally with my implementation